.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,32 +1,49 @@
-Fixes #
-
-### Changes proposed in this Pull Request
+Resolves #
 
+## Proposed Changes
 *
 
-### Testing instructions
-
+## Testing Instructions
 <!--
-If the functionality of this PR is behind a feature flag, please include the
-following testing step, using the correct name for the feature flag. (If you
-aren't sure, just ignore this step)
-
-* Enable the feature flag: `add_filter( 'sensei_feature_flag_{THE_FLAG_NAME}', '__return_true' );`
+Add as many details as possible to help others reproduce the issue and test the fix.
+"Before / After" screenshots can also be very helpful when the change is visual.
 -->
-*
 
+1.
+2.
+3.
+
+## New/Updated Hooks
 <!-- Add the following only if there are new/updated actions or filters. Please provide a brief description of what they do and any arguments they may take. Be sure to also add the "Hooks" label to this PR. -->
-### New/Updated Hooks
 
 *
 
+## Deprecated Code
 <!-- Add the following only if there is any code that is being deprecated. Please list the replacement function or hook that should be called instead, if applicable. Be sure to also add the "Deprecation" label to this PR. -->
-### Deprecated Code
 
 *
 
-<!--
-Helpful tips for screenshots:
-https://en.support.wordpress.com/make-a-screenshot/
--->
-### Screenshot / Video
+## Pre-Merge Checklist
+<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.
+
+Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
+- [ ] PR title and description contain sufficient detail and accurately describe the changes
+- [ ] Acceptance criteria is met
+- [ ] Decisions are publicly documented
+- [ ] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
+- [ ] All strings are translatable (without concatenation, handles plurals)
+- [ ] Data is [sanitized](https://developer.wordpress.org/apis/security/sanitizing/) / [escaped](https://developer.wordpress.org/apis/security/escaping/)
+- [ ] Follows our [naming conventions](https://wp.me/P6rkRX-4oA)
```

Comment:
Just out of curiosity, should we have non-public URLs here? For example, in case a contributor wants to follow the correct naming convention and tries to check the documentation

Reply:
I switched them to use our shorthand version in https://github.com/Automattic/sensei/pull/6542/commits/123acb9084d7c73f015a68bf253bbae4d4f47f80 so that they'll only appear as links for a11ns. We could possibly consider adding these docs to a public place, but I'm not planning to do that as part of this work.